grafana-plugin/src/containers/TemplatePreview/TemplatePreview.tsx

```diff
@@ -83,68 +86,87 @@ const TemplatePreview = observer((props: TemplatePreviewProps) => {
-              {conditionalMessage(conditionalResult.value === 'True')}
-            </VerticalGroup>
-          )}
-        </Text>
-      ) : (
-        <>
-          {templateName.includes('image') ? (
-            <div className={cx('image-result')}>
-              <img
-                src={result.preview}
-                onError={(e) => {
-                  const target = e.target as HTMLImageElement;
-                  target.alt = result.preview || 'No image found';
+  function renderResult() {
+    switch (templateType) {
+      case 'html': {
+        return renderHtmlResult();
+      }
+      case 'image': {
+        return renderImageResult();
+      }
+      case 'boolean': {
+        return renderBooleanResult();
+      }
+      case 'plain': {
+        return renderPlainResult();
+      }
+      default: {
+        return renderPlainResult();
+      }
+    }
+  }
+  function renderBooleanResult() {
+    return (
+      <Text type={conditionalResult.value === 'True' ? 'success' : 'danger'}>
+        {conditionalResult.value === 'True' ? (
+          <VerticalGroup>
+            <HorizontalGroup>
+              <Icon name="check" size="lg" /> {conditionalResult.value}
+            </HorizontalGroup>
+            {conditionalMessage(conditionalResult.value === 'True')}
+          </VerticalGroup>
+        ) : (
+          <VerticalGroup>
+            <HorizontalGroup>
+              <Icon name="times-circle" size="lg" />
+              <div
+                className={cx('message')}
+                dangerouslySetInnerHTML={{
+                  __html: sanitize(result.preview),
                 }}
               />
-            </div>
-          ) : templateName.includes('web_') ? (
-            <div
-              className={cx('message')}
-              dangerouslySetInnerHTML={{
-                __html: sanitize(result.preview),
-              }}
-            />
-          ) : (
-            // <Text className={cx('display-linebreak')}>
-            //   {sanitize(result.preview)}
-            // </Text>
-            <div
-              className={cx('message', 'pre-wrap')}
-              dangerouslySetInnerHTML={{
-                __html: sanitize(result.preview),
-              }}
-            />
-          )}
-        </>
-      )}
-    </>
-  ) : (
-    <LoadingPlaceholder text="Loading..." />
-  );
+            </HorizontalGroup>
+            {conditionalMessage(conditionalResult.value === 'True')}
+          </VerticalGroup>
+        )}
+      </Text>
+    );
+  }
+  function renderHtmlResult() {
+    return (
+      <div
+        className={cx('message')}
+        dangerouslySetInnerHTML={{
+          __html: sanitize(result.preview),
+        }}
+      />
+    );
+  }
+  function renderPlainResult() {
+    return (
+      <div
+        className={cx('message', 'display-linebreak')}
+        dangerouslySetInnerHTML={{
+          __html: sanitize(result.preview),
+        }}
+      />
+    );
+  }
```

Comment:
If just the class differs we could reuse the same block for both html and plain

```jsx
function renderHTMLOrPlainResult() {
    return (
      <div
        className={cx('message', { 'display-linebreak': templateType === 'plain' })}
        dangerouslySetInnerHTML={{
          __html: sanitize(result.preview),
        }}
      />
    );
  }
```

Reply:
I also use it as a fallback option
https://github.com/grafana/oncall/pull/2304/files#diff-3c6030802a721e3eb3d2709e36949200052f283d912f7add6135ff2ba0fd7290R100